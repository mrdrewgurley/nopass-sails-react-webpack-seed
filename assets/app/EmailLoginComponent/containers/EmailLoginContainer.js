import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import * as EmailLoginActions from '../actions/EmailLoginActions'
import EmailLoginButtonComponent from '../components/EmailLoginButtonComponent'
import EmailLoginInputComponent from '../components/EmailLoginInputComponent'

const EmailLoginContainer = props => (
  <div>
    <EmailLoginInputComponent
      Actions={props.Actions}
      EmailLoginInputReducer={props.EmailLoginInputReducer}
    /><br /><br />
    <EmailLoginButtonComponent
      Actions={props.Actions}
      EmailLoginButtonReducer={props.EmailLoginButtonReducer}
    />
  </div>
)

EmailLoginContainer.propTypes = {
  Actions: PropTypes.object.isRequired,
  EmailLoginButtonReducer: PropTypes.object.isRequired,
  EmailLoginInputReducer: PropTypes.object.isRequired
}

export default connect(
  state => ({ ...state }),
  dispatch => ({ Actions: bindActionCreators(EmailLoginActions, dispatch) })
)(EmailLoginContainer)
