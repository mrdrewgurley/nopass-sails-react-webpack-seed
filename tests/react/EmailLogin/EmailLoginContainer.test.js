import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import EmailLoginContainer from '../../../assets/app/EmailLoginComponent/containers/EmailLoginContainer'
import Store from '../../../assets/app/EmailLoginComponent/stores/EmailLoginStore'

/**
 * React.EmailLoginContainer
 * @desc Tests for the Email Login Container
 */
describe("Email Login Container", () => {
  it("renders with proper defaults", () => {
    const EmailLogin = renderer
      .create(<Provider store={Store()}>
        <EmailLoginContainer />
      </Provider>)
      .toJSON()

    expect(EmailLogin).toMatchSnapshot()
  })
})
