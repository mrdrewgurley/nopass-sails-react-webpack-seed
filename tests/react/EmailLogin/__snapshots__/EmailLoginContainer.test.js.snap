// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Login Container renders with proper defaults 1`] = `
<div>
  <div
    className="uk-inline"
    id="EmailLoginInput"
  >
    <span
      className="uk-form-icon"
      uk-icon="icon: user"
    />
    <input
      className="uk-input enabled"
      disabled={false}
      id="form-input-email"
      required={true}
    />
    <label
      htmlFor="form-input-email"
    >
      Email Address
    </label>
  </div>
  <br />
  <br />
  <div
    className="uk-button-group"
  >
    <button
      className="uk-button enabled"
      disabled={false}
      onClick={[Function]}
    >
      SEND LOGIN EMAIL
    </button>
  </div>
</div>
`;
